celery/backends/mongodb.py

```diff
@@ -157,6 +157,10 @@ def _get_connection(self):
             # don't change self.options
             conf = dict(self.options)
             conf['host'] = host
+            if self.user:
```

Comment:
Good observation. Adding the [library documentation link](https://api.mongodb.com/python/current/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient) for reference (`Parameters -> Authentication` section).